### PR TITLE
Multichannel features

### PR DESCRIPTION
**Pull request by dafaronbi**
#### Reference Issue 
Implements #1130 

#### What does this implement/fix? Explain your changes.
work in progress for features module

### CODE REVIEW

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:728]: this sum is probably on the wrong axis

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:951]: shape check is probably on the wrong axis

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:1173]: i think this might need to be -1 here, not -2

File: librosa/feature/spectral.py
Comment (bmcfee) [reply librosa/feature/spectral.py]: Re-upping this: i think this cumsum should be on `axis=-2`

Comment (bmcfee) [reply librosa/feature/spectral.py]: similarly, the usage of `total_energy` below should probably move to `total_energy[..., -1, :]`, and the nanmin axis should be -2.

File: tests/test_multichannel.py
Comment (bmcfee) [inline tests/test_multichannel.py:362]: This one fails because of the power_to_db calculation in mfcc when the input spectrogram is not provided.
For now, I think we can bypass this test.  Rather than delete it, let's add `@pytest.mark.skip(reason='power_to_db leaks information across channels')` on this test, and then we can come back and fix it later (if we provide API hooks that get us around it).

File: tests/test_multichannel.py
Comment (bmcfee) [inline tests/test_multichannel.py:372]: It's okay to keep prints in tests during development, but they should be removed prior to pushing.

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:205]: centroid LGTM

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:381]: Bandwidth LGTM; that broadcast subtraction still feels a little awkward to me, but I can live with it for now.

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:742]: I wonder if we could bypass this `expand_dims` call by instead doing
```python
threshold = roll_percent * total_energy[...,-1:,:]
```
(Note the `-1:` instead of `-1` for the penultimate axis slice.)

I bring this up mainly because I'm worried we might run aground of some deprecated behavior in expand_dims described here: https://numpy.org/doc/stable/reference/generated/numpy.expand_dims.html

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:877]: flatness LGTM

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:984]: rms lgtm

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:1132]: I'll deal with polyfeatures separately after we merge this PR

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:1194]: ZCR LGTM

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:1354]: chroma_stft basically LGTM.  There is a subtlety here around `estimate_tuning` which I've already noted in this comment: https://github.com/librosa/librosa/pull/1351#issuecomment-894249191

We can leave this is for now, but I'll have to come back to it once we decide on how to handle tuning in multichannel chroma.

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:2050]: mfcc LGTM for now; we may want to revisit the idea of exposing power_to_db params at the API here.

File: librosa/feature/spectral.py
Comment (lostanlen) [inline librosa/feature/spectral.py:165]: remove these comments and empty lines?

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:1779]: Test is failing because we lost the normalization on chroma

File: librosa/feature/spectral.py
Comment (bmcfee) [reply librosa/feature/spectral.py:165]: I'm not too worried about empty lines / formatting right now.  We'll `black` the code before merging into main.

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:1490]: chroma_cqt lgtm

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:1661]: CENS lgtm

File: librosa/feature/spectral.py
Comment (bmcfee) [inline librosa/feature/spectral.py:1769]: tonnetz LGTM

File: librosa/feature/utils.py
Comment (bmcfee) [inline librosa/feature/utils.py:301]: stack LGTM

**Review by bmcfee — APPROVED** [review]

**Comment by bmcfee** [comment]
This is :shipit: IMO!  There are still some rough edges to hammer out, but they depend on other changes to make elsewhere in the library. I'll merge with #1351 once CI passes.

**Pull request merged**
